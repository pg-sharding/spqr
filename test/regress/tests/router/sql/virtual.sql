
SELECT 1;

SELECT 1, 2, 3;

SELECT 'postgres is cool';

SELECT pg_is_in_recovery();

SELECT current_setting('transaction_read_only');

SELECT pg_is_in_recovery(), 1, 'kek';

SELECT CURRENT_USER;

--- XXX: support
--- SELECT 1,2,3 UNION ALL SELECT 2,3,4;

-- XXX:support
--SHOW transaction_read_only;

-- XXX: support ignore patterns to test this
-- SELECT now(), 1;
