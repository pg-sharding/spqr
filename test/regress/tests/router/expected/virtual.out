SELECT 1;
 ?column? 
----------
        1
(1 row)

SELECT 1, 2, 3;
 ?column? | ?column? | ?column? 
----------+----------+----------
        1 |        2 |        3
(1 row)

SELECT 'postgres is cool';
     ?column?     
------------------
 postgres is cool
(1 row)

SELECT pg_is_in_recovery();
 pg_is_in_recovery 
-------------------
 f
(1 row)

SELECT __spqr__is_ready();
 __spqr__is_ready 
---------------
 t
(1 row)

SELECT current_setting('transaction_read_only');
 current_setting 
-----------------
 f
(1 row)

SELECT pg_is_in_recovery(), 1, 'kek';
 pg_is_in_recovery | ?column? | ?column? 
-------------------+----------+----------
 f                 |        1 | kek
(1 row)

SELECT CURRENT_USER;
 ?column? 
----------
 regress
(1 row)

set __spqr__preferred_engine to v2;
select __spqr__shards();
 shard name 
------------
 sh1
 sh2
 sh3
 sh4
(4 rows)


--- XXX: support
--- SELECT 1,2,3 UNION ALL SELECT 2,3,4;
-- XXX:support
--SHOW transaction_read_only;
-- XXX: support ignore patterns to test this
-- SELECT now(), 1;
